server/src/grpc/remote_engine_service/mod.rs

```diff
@@ -348,6 +351,11 @@ async fn handle_write(ctx: HandlerContext, request: WriteRequest) -> Result<Writ
             code: StatusCode::Internal,
             msg: format!("fail to write table, table:{:?}", write_request.table),
         })?;
+    REMOTE_ENGINE_GRPC_HANDLER_COUNTER_VEC
+        .write
+        .failed
+        .inc_by((row_count - affected_rows) as u64);
```

Comment:
If it reaches here, there is no failure happens. Maybe you should collect the metrics for L353 when the returned value is a error.